core/src/test/java/com/opensymphony/xwork2/interceptor/CSRFPreventionInterceptorTest.java

```diff
@@ -0,0 +1,152 @@
+package com.opensymphony.xwork2.interceptor;
+
+
+import com.opensymphony.xwork2.ActionContext;
+import com.opensymphony.xwork2.XWorkTestCase;
+import com.opensymphony.xwork2.mock.MockActionInvocation;
+import org.apache.struts2.ServletActionContext;
+import org.springframework.mock.web.MockHttpServletRequest;
+
+import java.util.Arrays;
+import java.util.List;
+
+// TODO Vary headers tests
```

Comment:
would be great to have an integration test too where we set up a few interceptors and check that the chain gets interrupted when an attack reaches our interceptor!